cirrus/google_cloud/manager.py

```diff
@@ -115,12 +115,43 @@ def log_backoff_giveup(details):
 
 
 def _is_handled_exception(e):
```

Comment:
the current name is confusing, because like rate limiting exceptions are technically 'not handled', let's rename this to something like `exceptions_without_retry` and add docstring to explain that we don't want to retry for errors we raise (CirrusError) and google errors that are not temporory and not recoverable by retry